src/services/validators/createServiceDto.validator.ts

```diff
@@ -0,0 +1,28 @@
+import { IValidator, ValidatedResponse } from '../../shared/validators/validtor.interface';
+import { CreateServiceRequestDto } from '../models/dtos/service/request/create-service.requrest.dto';
+
+export class CreateServiceDtoValidator implements IValidator<CreateServiceRequestDto> {
+    validate(obj: CreateServiceRequestDto): ValidatedResponse {
+        const errorMessages = [];
+        if (obj.name == null || obj.name.trim().length <= 0) {
+            errorMessages.push('name cannot be empty');
+        }
+
+        if (obj.location == null) {
+            errorMessages.push('location cannot be empty');
```

Comment:
Prefer error codes, like we used for DB_Error.OfflineError - Client can take the responsibility of translating the error code to error messages with localisation.

Reply:
apram paniklam dude
english problem :D